include/pybind11/cast.h

```diff
@@ -960,9 +960,14 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
 private:
     using caster_t = make_caster<type>;
     caster_t subcaster;
-    using subcaster_cast_op_type = typename caster_t::template cast_op_type<type>;
-    static_assert(std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value,
-            "std::reference_wrapper<T> caster requires T to have a caster with an `T &` operator");
+    using reference_t = type&;
```

Comment:
This could still be collapsed where it's used, to be shorter and more transparent (which was why I suggested `type &` instead of `add_lvalue_reference<type>::type`). But if that's the only thing left to change, I'm fine with this as well, and you don't need to add yet another commits, AFAIC.